hooks/useListsData.ts

```diff
@@ -0,0 +1,144 @@
+import { useState, useEffect } from 'react';
+import { useAuth } from '../app/auth/context/AuthContext';
+import type { Lists, ListType } from '../ambient';
+import { LISTS, MOVIES } from '../constants';
+
+type UseFetchListsReturn = {
+  lists: ListType[];
+  sharedLists: ListType[];
+  loading: boolean;
+  error?: Error;
+};
+
+export const useFetchLists = (): UseFetchListsReturn => {
```

Comment:
I'm not really familiar with the convention here, but isn't it better if we keep `hook` and `file name` the same?